test/Transactions/Orleans.Transactions.Tests/Runners/TransactionRecoveryTestsRunner.cs

```diff
@@ -44,21 +50,26 @@ public virtual Task TransactionWillRecoverAfterRandomSiloUnGracefulShutdown(stri
         protected virtual async Task TransactionWillRecoverAfterRandomSiloFailure(string transactionTestGrainClassName, bool gracefulShutdown)
         {
             const int grainCount = 100;
-            var txGrains = Enumerable.Range(0, grainCount)
-                .Select(i => RandomTestGrain(transactionTestGrainClassName))
+            int index = 0;
+            Func<int> getIndex = () => { return index++; };
+            List<ExpectedGrainActivity> txGrains = Enumerable.Range(0, grainCount)
+                .Select(i => new ExpectedGrainActivity { Grain = RandomTestGrain<ITransactionalBitArrayGrain>(transactionTestGrainClassName) })
                 .ToList();
-            var txSucceedBeforeInterruption = await AllTxSucceed(txGrains);
+            var txSucceedBeforeInterruption = await AllTxSucceed(txGrains, getIndex());
+            await ValidateResults(txGrains);
             this.logger.LogInformation($"Tx succeed before interruption : {txSucceedBeforeInterruption}");
             if(gracefulShutdown)
-                this.testCluster.StopSilo(this.testCluster.Silos[this.seed.Next(this.testCluster.Silos.Count)]);
+                this.testCluster.StopSilo(this.testCluster.Silos[this.random.Next(this.testCluster.Silos.Count)]);
             else
-                this.testCluster.KillSilo(this.testCluster.Silos[this.seed.Next(this.testCluster.Silos.Count)]);
-            await TestingUtils.WaitUntilAsync(lastTry => CheckTxResult(txGrains, lastTry), RecoveryTimeout);
+                this.testCluster.KillSilo(this.testCluster.Silos[this.random.Next(this.testCluster.Silos.Count)]);
+            await TestingUtils.WaitUntilAsync(lastTry => CheckTxResult(txGrains, getIndex, lastTry), RecoveryTimeout);
+            output.WriteLine($"Performed {index} transactions");
```

Comment:
If we want to access `index` outside of `getIndex()`, then we need to use an array, like `index = new [] {0}` and access it as `index[0]`, otherwise we're accessing a copy.